### DISCUSSION TITLE
sortby and filterby doesn't work

### DISCUSSION THREAD

**Discussion by spierjoe** [primefaces/community#1004] [PrimeFaces]
After change the Primeface-Version, sortby and filterby in column doesn't work.

```
    <p:dataTable id="lvzs" var="mylvz" value="#{wertelistenBean.laenderCodeListItems}"
            emptyMessage="Kein L&auml;nderverzeichnis vorhanden"
            dynamic="true" rowKey="#{mylvz.lkz}"
			paginator="true" rows="10">
	<f:facet name="header"><h:outputText value="Liste L&auml;nderverzeichnis"/></f:facet>

        <p:column headerText="L&auml;nderID" 
                sortBy="#{mylvz.lkz}">
            <h:outputText value="#{mylvz.lkz}"/>
        </p:column>
        
        <p:column headerText="Land"
                sortBy="#{mylvz.land}" filterBy="#{mylvz.land}" filterMatchMode="startsWith">
            <h:outputText value="#{mylvz.land}"/>
        </p:column>
    </p:dataTable>
```

before:
PrimeFaces 6.2., Extensions 6.2.11
Java SE 8

new:
PrimeFaces 13.0.5-jakarta, Extensions 13.0.5-jakarta
jakarta-faces 3.0.5
Java SE 17
spring-boot 3.1.4

**Comment by melloware** [comment]
A couple of things.

1. There is no such table property as `dynamic="true"` not sure what you are trying to do there.
2.  you can greatly simply your column for sorting and filtering by changing the following.

Yours:
```xml
 <p:column headerText="Land"
                sortBy="#{mylvz.land}" filterBy="#{mylvz.land}" filterMatchMode="startsWith">
            <h:outputText value="#{mylvz.land}"/>
</p:column>
```

TO:
```xml
<p:column headerText="Land" field="land" filterMatchMode="startsWith"/>
```

See Field example: https://www.primefaces.org/showcase/ui/data/datatable/field.xhtml?jfwid=eca72

Also check the showcase all the sorting and filtering examples are working:

https://www.primefaces.org/showcase/ui/data/datatable/sort.xhtml?jfwid=eca72

https://www.primefaces.org/showcase/ui/data/datatable/filter.xhtml?jfwid=eca72

**Reply by spierjoe** [reply]
The switch to "field" has only had limited success.

Interesting is:
If I start the SprinBoot application from the Eclipse development environment, sorting and filtering work perfectly.
If I start the SprinBoot application from a generated .war file, sorting and filtering do not work. As if something was missing, or an event was intercepted or handled differently.

**Reply by melloware** [reply]
Can you check F12 console for JavaScript errors?  its almost like you have a Spring Boot filter blocking AJAX traffic or something in Production Mode.

**Reply by spierjoe** [reply]
I adapted the code accordingly.
```
        <p:column headerText="L&auml;nderID" field="lkz" filterable="false">
            <h:outputText value="#{mylvz.lkz}"/>
        </p:column>
        
        <p:column headerText="Land" field="land" >
            <h:outputText value="#{mylvz.land}"/>
        </p:column>
```


No javascript errors are displayed.


A request is sent.
![Bildschirmfoto 2024-01-26 um 14 43 09](https://github.com/primefaces/community/assets/86762699/8fa9bf7f-1d1f-43f8-aee5-bea4758db25d)


An answer will be received. However, the unchanged representation.
![Bildschirmfoto 2024-01-26 um 14 44 20](https://github.com/primefaces/community/assets/86762699/22baaa3f-842d-4154-936b-8b987b006835)

Is there perhaps still a problem with the interaction with Jakarta?

**Reply by melloware** [reply]
I wouldn't think so I use Jakarta Faces every day without issue.  Most likely a Spring Boot related issue as that is the one outlier in your setup. I can't reproduce on the Showcase or with our [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.

**Reply by melloware** [reply]
Also don't you want Jakarta Faces 4.0.5 not 3.0.5 ?

**Reply by spierjoe** [reply]
Your PrimeFaces test project works with javax and not with jakarta.
Do you have a test project based on jakarta EE-9 for me? 
Like for example with:
  ```
      <dependency>
            <groupId>org.primefaces</groupId>
            <artifactId>primefaces</artifactId>
            <classifier>jakarta</classifier>
            <version>13.0.5</version>
        </dependency>

```
It is then jakarta Faces 3.0.5. Jakarta Faces 4.0.5 works with jakarta EE-10

**Reply by melloware** [reply]
Yep of course we have a Jakarta branch all set up and working: https://github.com/primefaces/primefaces-test/tree/jakarta

**Comment by spierjoe** [comment]
The error occurs when using `mvn clean package` to generate a **.war** file. What is blocking springboot is unfortunately unknown.
We changed the code a little. If a **.jar** file is generated with `mvn clean package`, the errors do not occur.

**Reply by melloware** [reply]
OK sounds like a packaging issue on your end.
